assets/tailwind.config.js

```diff
@@ -4,6 +4,7 @@
 const plugin = require("tailwindcss/plugin");
 const fs = require("fs");
 const path = require("path");
+// const colors = require("./colors"); // QQ: @ks0m1c_dharma any idea why if i uncomment this, the tailwind plugin config gets broken?
```

Comment:
<a id="org94132b5"></a>

# FIXME odd tailwind config issue:

So after the annoyance about using css variables to define other css variables and how even after doing the hsl syntax, when using them in tailwind utility classes, it wasn&rsquo;t possible to add opacity classes to it,
[here&rsquo;s a related gh comment for it](https://github.com/shadcn-ui/ui/issues/805)

I decided to try using an external `colors.js` so that I can make my best friend, JS, do the magic.

Turns out this breaks the tailwindconfig file and some other import doesn&rsquo;t get its name resolved. I have no idea why, let me know if you know this.

```js
    Rebuilding...
    Error: Cannot find module 'tailwindcss/plugin'
    Require stack:
    - /Users/rtshkmr/Projects/vyasa/assets/tailwind.config.js
        at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
        at Function._resolveFilename (pkg/prelude/bootstrap.js:1955:46)
        at Function.resolve (node:internal/modules/cjs/helpers:108:19)
        at _resolve (/snapshot/tailwindcss/node_modules/jiti/dist/jiti.js:1:241025)
        at jiti (/snapshot/tailwindcss/node_modules/jiti/dist/jiti.js:1:243309)
        at /Users/rtshkmr/Projects/vyasa/assets/tailwind.config.js:4:16
        at jiti (/snapshot/tailwindcss/node_modules/jiti/dist/jiti.js:1:245784)
        at /snapshot/tailwindcss/lib/lib/load-config.js:37:30
        at loadConfig (/snapshot/tailwindcss/lib/lib/load-config.js:39:6)
        at Object.loadConfig (/snapshot/tailwindcss/lib/cli/build/plugin.js:135:49) {
      code: 'MODULE_NOT_FOUND',
      requireStack: [ '/Users/rtshkmr/Projects/vyasa/assets/tailwind.config.js' ]
    }
```

Using JS might be more convenient.
perhaps this functional fetching of colors might make work [as seen here](https://stackoverflow.com/questions/74521806/how-to-add-custom-opacity-values-to-colors-in-tailwind-css-themes/74719392#74719392)?
I&rsquo;m going to side-step this for now.